### PR TITLE
feat: Autocomplete result TAB scrolling

### LINKED ISSUE #115: Pressing TAB should scroll through possible options
## Currently

We cannot scroll through possible autocompleted options.

## Suggestion

Store the text meant to autocomplete, and use that text and go through a circular list of autocomplete results. Pressing tab moves us through this list.

### PR DESCRIPTION
**Pull request by tituschewxj**
Adds TAB scrolling through autocomplete result.

**Example 1**
Pressing TAB repeatedly: `mark nn/` should go through the all NUSNet IDs.

**Example 2**
Press TAB repeatedly: `mark nn/e1` should go through the all NUSNet IDs starting with `e1`.

Resolves #115

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/Trie.java [hunk 1]
```diff
@@ -110,6 +113,9 @@ public String findFirstWordWithPrefix(String prefix) {
             sb.append(c);
             current = current.getChild(c);
         }
+        System.out.println(prefix);
+        System.out.println(findAllWordsWithPrefix(prefix));
+
```
Comment (ForAeons) [inline src/main/java/seedu/address/commons/util/Trie.java:118]:
```suggestion

```

File: src/main/java/seedu/address/logic/autocomplete/AutoCompleteResult.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** The internal pointer to the current result */
+    private int currentIndex;
+
+    /**
+     * Creates an autocomplete result given a results list.
+     */
+    public AutoCompleteResult(List<String> results) {
+        this.results = results;
+        System.out.println(results);
```
Comment (ForAeons) [inline src/main/java/seedu/address/logic/autocomplete/AutoCompleteResult.java:19]:
```suggestion
```

**Review by ForAeons — APPROVED** [review]
Fantastic work! I never thought this feature could be implemented like this.

The feature works flawlessly on my machine and just remove the left over debugging lines and the rest LGTM

File: src/main/java/seedu/address/commons/util/Trie.java (hunk 1 above)
Comment (tituschewxj) [reply src/main/java/seedu/address/commons/util/Trie.java:118]: Thanks for catching this!

**Review by saiutkarsh33 — APPROVED** [review]
LGTM

**Pull request merged**